### PR TITLE
[Proposal] SwiftGen and Localization

### PR DESCRIPTION
**Pull request by emeseuk**
I concluded my findings related to how we can modularise the strings we have.
Please let me know what's your opinion about it.

Thank you ❤️

### CODE REVIEW

File: Cookbook/Proposals/SwiftGenAndLocalization.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
+# SwiftGen and Localization
+
+* Author: Emese Toth
+* Review Manager: TBD
+
+## Introduction
+
+In order to modularise our strings we need to triage the existing ones and separate them per feature and possibly mark the unused ones as deprecated. 
+The reason for this is that the next time we would like to create a new app, similarly to the Barzil one, we would only translate a subset of localization which is relevant to that new application.
```
Comment (Ben-Henshall) [inline Cookbook/Proposals/SwiftGenAndLocalization.md:9]:
```suggestion
The reason for this is that the next time we would like to create a new app (such as our SAAS initiative), we would only translate a subset of localization which is relevant to that new application.
```

File: Cookbook/Proposals/SwiftGenAndLocalization.md [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+One of the problems we’ll face is that we have a rule about not deleting old keys in Lokalise. So we’ll need to find a way to rename the existing keys we have in Lokalise so that we can still provide the modularisation we’re aiming for.
+
+# Possible solutions
+# 1 Key referencing
+
+One way could be using Lokalise’s key referencing feature. In Lokalise every key has a unique key_id, with these the translations can be linked to each other to allow the reuse of the existing translations in the new strings.
+Example of key referencing:
+```swift
+general_cancel = "Cancel"  //key_id_1
+payments_cancel_action = "[key_id_1] this action."  //key_id_2
```
Comment (Ben-Henshall) [inline Cookbook/Proposals/SwiftGenAndLocalization.md:42]: Who decides if a key can be referenced? The developer or the translator?

I'm worried that some languages have different rules that would prevent us from using this. e.g. the word for 'cancel' may be different based on the context, preventing us from using this key referencing.

Comment (emeseuk) [reply Cookbook/Proposals/SwiftGenAndLocalization.md:42]: This is just an example of how Lokalise intended to use the key referencing feature.
My idea is to use this to change the format of our current keys by
1. creating new keys and adding the content of the old keys there:
`framework_name.feature.error_message = "Something went wrong, please try again later"` 
2. referencing the key_id of this newly created key to the old keys:
`framework_name_feature_error_message` = [key_id of framework_name.feature.error_message ]

**Comment by emeseuk** [comment]
I'm closing this Proposal. Due to RN we won't need this change anymore.

**Pull request closed without merging**